### DISCUSSION TITLE
I大，求教🙏，我希望在同样是debian-based的系统TrueNAS上驱动dg1，以达到jellyfin解码的目的。

### DISCUSSION THREAD

**Discussion by ChingyimLee** [Icarusradio/intel-media-driver-dg1#22] [Q&A] [answered]
个人原因，不是很喜欢虚拟机，我的nas是truenas-scale
我参考了网上几位大神的教程，起草了给truenas-scale 24.04.2.2版本编译dg1驱动的过程，在我的设备，驱动可以安装，vainfo可以正常输出，但是jellyfin无法解码，会发生致命错误，且发生致命错误后，即便重启系统，vainfo输出也会报错，无法恢复。
我的过程，请您指教：
https://github.com/ChingyimLee/dg1ontruenas-scale
我不知到底是什么原因导致的，但是看起来整个驱动确实是装上了。

**Comment by ChingyimLee** [comment]
我没有学习过任何编程课程，即便是大学本科是工科化工，专业也没有c语言的必修课程，所以基础很薄弱，可以说我只是个硬件爱好者，有纰漏请您不吝赐教。

**Comment by Icarusradio** [comment, marked as answer]
TrueNAS Scale 我没有研究过，不太能帮助到你。
Intel 官方只给 Ubuntu 做了适配，不代表 debian 系的都能成功。我之前试过给 debian 装 intel-i915-dkms，编译模块时会报错。

**Reply by ChingyimLee** [reply]
I大好～我亲测，尝试用您编译的iHD驱动替换了我自己编译的iHD驱动后，在truenas可以直驱DG1。测试emby可以直接转码，且jellyfin替换内建驱动后，解码效率也和您和皮蛋熊大神测试的帧数大差不差。
这其中，我不太明白的就是，我起初也是按照相应步骤编译，编译安装过程中没有报错，编译的驱动反而不能用，用您编译出来的一替换就行了。

**Reply by Icarusradio** [reply]
> I大好～我亲测，尝试用您编译的iHD驱动替换了我自己编译的iHD驱动后，在truenas可以直驱DG1。测试emby可以直接转码，且jellyfin替换内建驱动后，解码效率也和您和皮蛋熊大神测试的帧数大差不差。 这其中，我不太明白的就是，我起初也是按照相应步骤编译，编译安装过程中没有报错，编译的驱动反而不能用，用您编译出来的一替换就行了。

我不太清楚你编译的具体步骤，所以不知道问题出在那里，不过能用就好。

**Reply by ChingyimLee** [reply]
https://github.com/ChingyimLee/dg1ontruenas-scale
不知您能否赏脸简单看一下，我中间大段引用了一些您的博文内容，望您见谅，仅作个人记录用，您看完我会改回private。

**Reply by Icarusradio** [reply]
我的编译环境是一台 Ubuntu 22.04 LTS 虚拟机，使用的软件版本 Releases 里面有写，带链接的就是从源代码编译的。
然后我编译的源代码是直接按照 tag 从 GitHub 上下载的，具体编译步骤可以看我那篇博客。我编译的时候似乎不需要 libva-utils。
如果你是 docker 运行 Jellyfin 那没必要把 media-driver 装到系统里，只需要替换 docker 内驱动就好。

**Reply by ChingyimLee** [reply]
感谢I大，希望您的驱动也能够长久更新下去，从省事角度讲，直接移植您编译好的最方便，但是我主要还是想自己搞明白一些，回头我自行使用ubuntu编译一下试试！
其实，我选择libva-utils的原因是我发现安装vainfo会从debian那边拉回一个libva，这就会导致libva的版本和media-driver版本不符，所以，我选择了自带vainfo的libva-utils。
此外，我将驱动安装到系统的原因主要是还想搞一搞emby，但是经过实际测试使用，emby对dg1的支持仿佛没有jellyfin好，能够驱动，但是帧数特别低，我现在主要还是使用jellyfin多一些。

**Reply by Icarusradio** [reply]
昨天和你参考文章其中一位作者交流过，Jellyfin 后面计划支持 6.8 内核自带的 xe 内核驱动，就不需要我编译给 i915 的特制驱动了。（[参考](https://github.com/Harry-Chen/blog-comments/issues/24#issuecomment-2368232585)）

**Reply by ChingyimLee** [reply]
好哒，不过实际上，truenas内核目前到6.6，下个大版本也许能高于6.8，拭目以待吧。如果真是这样反而方便了，原生支持。
